src/silo/zstdfasta/zstdfasta_table_reader.cpp

```diff
@@ -0,0 +1,112 @@
+#include "silo/zstdfasta/zstdfasta_table_reader.h"
+
+#include <cstddef>
+#include <fstream>
+#include <stdexcept>
+
+#include <fmt/format.h>
+#include <spdlog/spdlog.h>
+#include <duckdb.hpp>
+#include <duckdb/common/types/blob.hpp>
+
+#include "silo/common/fasta_format_exception.h"
+#include "silo/preprocessing/preprocessing_exception.h"
+#include "silo/zstdfasta/zstd_decompressor.h"
+
+silo::ZstdFastaTableReader::ZstdFastaTableReader(
+   duckdb::Connection& connection,
+   std::string_view table_name,
+   std::string_view compression_dict,
+   std::string_view where_clause,
+   std::string_view order_by_clause
+)
+    : connection(connection),
+      table_name(table_name),
+      where_clause(where_clause),
+      order_by_clause(order_by_clause),
+      decompressor(std::make_unique<ZstdDecompressor>(compression_dict)),
```

Comment:
Oh yeah, this was a little cumbersome to deal with. In short: Because the ZstdDecompressor uses zstd library components that are neither trivially copyable / moveable nor are easy to write move constructors for (I tried, but many months ago, when introducing Zstd(De)compressors). I want the table reader to be moveable however. A unique ptr makes this possible, as it is moveable (a moved ptr stays a valid pointer)

Reply:
I see, thanks.